drivers/dma/dma_mcux_lpc.c

```diff
@@ -374,6 +374,8 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 		data = DEV_CHANNEL_DATA(dev, virtual_channel);
 	}
 
+	memset(data, 0, sizeof(struct channel_data));
```

Comment:
This `memset` call is placed too far into the `configure` implementation- we will wipe out the DMA handle settings applied by `DMA_CreateHandle`. Additionally, since we only call `DMA_CreateHandle` when "allocating" a virtual channel, and then reuse the virtual channel, shouldn't we avoid `memset` entirely here? Maybe the best approach would be to reset parts of the channel data that would logically be reconfigured, IE:
- `num_of_descriptors`
- `curr_descriptor`
- `descriptors_queued`
- `width`
- `busy`
- `dma_callback`
- `dir`

Reply:
@danieldegrasse, thank you. I have updated the implementation.